Core/Inc/ws2812.hpp

```diff
@@ -18,24 +28,21 @@
  * Please see led_buffer_init() function for the reason for the 2 extra frames.
  * 1 bit in a frame is mapped to 1 byte in the output buffer.
  */
-#define BUFF_SIZE (NUM_LEDS + 2) * 24
-
-extern uint8_t out_buf[BUFF_SIZE]; //Byte buffer of values to be transferred to the capture compare register (CCR).
+#define BUFF_SIZE (NUM_LEDS + 2) * 24  // We add 24 to the end so that we have 24 (one LED) bits of 0
```

Comment:
I would wrap BUFF_SIZE in another set of brackets:
```suggestion
#define BUFF_SIZE ((NUM_LEDS + 2) * 24)  // We add 24 to the end so that we have 24 (one LED) bits of 0
```

This prevents an ambiguous case where we use buff size and order precedence or formatting/syntax gets the best of us. For example:

 `BUFF_SIZE >>4` or similar, and we end up with `(NUM_LEDS + 2) * (24 >> 4)`.

Reply:
Do  double check  that our code works properly after this change though.

My personal preference is also to use something called a [constexpr ](https://en.cppreference.com/w/cpp/language/constexpr) although I won't force you to make that change here.